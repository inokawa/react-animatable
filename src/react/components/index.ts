export { TransitionGroup } from "./TransitionGroup";
export type { TransitionGroupProps } from "./TransitionGroup";
