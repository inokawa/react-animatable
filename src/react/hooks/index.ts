export * from "./useAnimation";
export * from "./useAnimationFunction";
export * from "./useTransitionAnimation";
export * from "./useScrollTimeline";
export * from "./useViewTimeline";
