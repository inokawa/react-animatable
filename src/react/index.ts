export * from "./components/index.js";
export * from "./hooks/index.js";
export * from "./types/index.js";
