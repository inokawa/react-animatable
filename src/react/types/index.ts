import {
  GetKeyframeFunction,
  TypedKeyframe,
  TypedKeyframeEffectOptions,
} from "../../core";
import { SCROLL_TIMELINE, VIEW_TIMELINE } from "./internal";

export interface AnimationOptions extends TypedKeyframeEffectOptions {
  timeline?: TimelineDefinition;
}

export type AnimationDefinition<Args> = [
  keyframe: TypedKeyframe | TypedKeyframe[] | GetKeyframeFunction<Args>,
  options?: AnimationOptions
];

export interface TimelineDefinition {
  (el: Element | null): void;
  /**
   * @private
   */
  _type: typeof SCROLL_TIMELINE | typeof VIEW_TIMELINE;
  /**
   * @private
   */
  _get: () => AnimationTimeline;
}

export type ScrollTimelineAxis = "block" | "inline" | "y" | "x";

export type ViewTimelineInset = "auto" | (string & {});
