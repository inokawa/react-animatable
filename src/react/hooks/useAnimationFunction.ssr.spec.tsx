/**
 * @jest-environment node
 */
import { it, describe, expect } from "@jest/globals";
import { useEffect, useState } from "react";
import { renderToString, renderToStaticMarkup } from "react-dom/server";
import {
  useAnimationFunction,
  ComputedTimingContext,
} from "./useAnimationFunction";

describe("SSR", () => {
  const Comp = () => {
    const [value, setValue] = useState(0);
    const animate = useAnimationFunction(
      (ctx: ComputedTimingContext) => {
        setValue(ctx.progress);
      },
      {
        duration: 600,
        easing: "ease-in-out",
      }
    );
    useEffect(() => {
      animate.play();
    }, []);
    return <div>{value}</div>;
  };

  it("should succeed with renderToString", () => {
    expect(renderToString(<Comp />)).toMatchSnapshot();
  });

  it("should succeed with renderToStaticMarkup", () => {
    expect(renderToStaticMarkup(<Comp />)).toMatchSnapshot();
  });
});
