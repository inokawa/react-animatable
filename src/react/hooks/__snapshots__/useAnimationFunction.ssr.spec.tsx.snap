// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR should succeed with renderToStaticMarkup 1`] = `"<div>0</div>"`;

exports[`SSR should succeed with renderToString 1`] = `"<div>0</div>"`;
